apps/meteor/server/models/raw/LivechatInquiry.ts

```diff
@@ -90,4 +101,93 @@ export class LivechatInquiryRaw extends BaseRaw<ILivechatInquiryRecord> implemen
 			{ $unset: { locked: 1, lockedAt: 1 } },
 		);
 	}
+
+	async getCurrentSortedQueueAsync({
+		_id,
+		department,
+	}: {
+		_id: string;
+		department: string;
+	}): Promise<Pick<ILivechatInquiryRecord, '_id' | 'rid' | 'name' | 'ts' | 'status' | 'department'> & { position: number }> {
+		const filter = [
+			{
+				$match: {
+					status: 'queued',
+					...(department && { department }),
+				},
+			},
+			{ $sort: await this.getSortingQuery() },
+			{
+				$group: {
+					_id: 1,
+					inquiry: {
+						$push: {
+							_id: '$_id',
+							rid: '$rid',
+							name: '$name',
+							ts: '$ts',
+							status: '$status',
+							department: '$department',
+						},
+					},
+				},
+			},
+			{
+				$unwind: {
+					path: '$inquiry',
+					includeArrayIndex: 'position',
+				},
+			},
+			{
+				$project: {
+					_id: '$inquiry._id',
+					rid: '$inquiry.rid',
+					name: '$inquiry.name',
+					ts: '$inquiry.ts',
+					status: '$inquiry.status',
+					department: '$inquiry.department',
+					position: 1,
+				},
+			},
+		] as UpdateFilter<ILivechatInquiryRecord>[];
+
+		// To get the current room position in the queue, we need to apply the next $match after the $project
+		if (_id) {
+			filter.push({ $match: { _id } });
+		}
+
+		return this.col.aggregate(filter).toArray() as unknown as Promise<
```

Comment:
.aggregate should accept a type parameter to avoid having the `as unknown as xxx` part 👀